dask/array/random.py

```diff
@@ -217,6 +217,11 @@ def choice(self, a, size=None, replace=True, p=None, chunks=None):
                 size = (size,)
 
             chunks = normalize_chunks(chunks, size, dtype=np.float64)
+            if not replace and len(chunks[0]) > 1:
```

Comment:
Why only on the first axis? Wouldn't this also fail for chunking on other axes?

Reply:
You're correct that it would also fail for multiple chunks on other axes. But `da.choice` only works with 1-D arrays

https://github.com/dask/dask/blob/16957a92d9c5f2a7fce2a494b0c9c9ec11a13652/dask/array/random.py#L188-L189

that's why I chose to check only `chunks[0]`.

Reply:
OK :)